filter.lua

```diff
@@ -112,32 +114,31 @@ local function filter(outbox)
 
 				local message = format('domain="%s",dport="%d",action="%s",reason="%s"',
 					domain, dport, action_name[verdict.action], verdict.reason)
-				outbox.notify(message)
+				outbox("notify", message)
 
 				if verdict.action == action.DROP then
-					outbox.reply(reply[dport] .. ":" .. tostring(packet))
+					outbox("reply", reply[dport] .. "|" .. tostring(packet))
 				end
 			end
 			return verdict.action
 		end
 	end
 end
 
-local function sender(action, queue)
-	return function (...)
-		local outbox = mailbox.outbox(queue)
-		local ok, err = pcall(outbox.send, outbox, ...)
+local function sender(queue, event)
+	local outbox = mailbox.outbox(queue, event)
+
+	return function (action, message)
+		local ok, err = pcall(outbox.send, outbox, action .. '|' .. message)
 		if not ok then
 			log("failed to %s: %s", action, err)
 		end
 	end
 end
 
-local function attacher(notify, reply)
-	local outbox = {
-		notify = sender('notify', notify),
-		reply = sender('reply', reply),
```

Comment:
I would keep the old scheme.. having two closures.. the overall code would be clearer this way, IMO.. we just need to have action being a argument of the outer function (sender) instead of the inner closure (return function () ... end)